predicators/envs/narrow_passage.py

```diff
@@ -0,0 +1,490 @@
+"""Toy environment for testing option learning."""
+
+from typing import ClassVar, Dict, Iterator, List, Optional, Sequence, Set
+
+import matplotlib
+import matplotlib.pyplot as plt
+import numpy as np
+from gym.spaces import Box
+
+from predicators import utils
+from predicators.envs import BaseEnv
+from predicators.settings import CFG
+from predicators.structs import Action, Array, GroundAtom, Object, \
+    ParameterizedOption, Predicate, State, Task, Type
+from predicators.utils import Rectangle, _Geom2D
+
+
+class NarrowPassageEnv(BaseEnv):
+    """An environment where a 2D point mass robot must reach a static 2D point
+    by passing through a narrow passage, or by opening a door and passing
+    through a wider passageway.
+
+    The action space is 1D, denoting the angle of movement. The
+    magnitude of the movement is constant. The point is considered
+    touched if the distance between the center of the robot and the
+    center of the target point is less than a certain threshold, which
+    is greater than the action magnitude.
+
+    Based on the TouchPoint and Doors environments.
+    """
+    x_lb: ClassVar[float] = 0.0
+    x_ub: ClassVar[float] = 1.0
+    y_lb: ClassVar[float] = 0.0
+    y_ub: ClassVar[float] = 1.0
+
+    robot_radius: ClassVar[float] = 0.1 / 2
+    target_radius: ClassVar[float] = 0.1 / 2
+    door_x_pos: ClassVar[float] = 0.2
+    door_width_padding: ClassVar[float] = 0.075
```

Comment:
we can delay this until later, but we may want to be able to change `door_width_padding` and/or `passage_width_padding` via command line, i.e., make them settings in settings.py

Reply:
Would prefer to delay this until it is needed to keep all current code in the same place for now